### PR TITLE
Add controllers spec

### PR DESCRIPTION
**Pull request by rugwizangoga**
# Blog app - controllers specs

- Created a Request spec file for all your controllers.
- Made sure that for each action there is the following checks:

```
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.
```

### CODE REVIEW

File: spec/requests/posts_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      get user_posts_path(user)
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the index template' do
+      user = User.create(name: 'jane', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Benin.',
+                         posts_counter: 0)
+      get user_posts_path(user)
+      expect(response).to render_template(:index)
+    end
```
Comment (anagudelogu) [inline spec/requests/posts_controller_spec.rb:17]: - [ ] According to the requirements:

Screenshot from 2023-08-02 09-29-25

This test suite is missing the test of the `response.body`. Please make sure to include it! 🤓 🔧

File: spec/requests/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      get user_path(user.id)
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'renders the show template' do
+      user = User.create(name: 'paul', photo: 'https://pbs.twimg.com/media/FfhkDtjWQAMYNrc.jpg:large', bio: 'Teacher from Burundi.',
+                         posts_counter: 0)
+      get user_path(user.id)
+      expect(response).to render_template(:show)
+    end
```
Comment (anagudelogu) [inline spec/requests/users_controller_spec.rb:17]: - [ ] The same happens here, the `response.body` test is not present. Please make sure to include it! 🤓 🔧

File: spec/requests/posts_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
```
Comment (anagudelogu) [inline spec/requests/posts_controller_spec.rb:3]: - [ ] Also, another thing I noticed is that you have an unexpected nesting in both your tests for `Posts` and `Users`:

image

Which, in code, looks like this:

```rb
describe 'GET #index' do
  ...
  describe 'GET #show' do
    ...
  end
end
```

Please consider fixing this structure to something like this:

```rb
describe 'GET #index' do
  ...
end

describe 'GET #show' do
  ...
end
```

Remember that both blocks are testing different requests, so it makes sense not to nest them! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @rugwizangoga, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by julie-ify — APPROVED** [review]
Hi @rugwizangoga,

## Highlights!
- Good code organization and tests ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**